### PR TITLE
chore: add test variant for credential reveal

### PR DESCRIPTION
**Pull request by gantunesr**
## **Description**

This PR adds a variant to the "Hold to reveal" button, this variant is a simple button that will be present in test builds to enable the E2E development of the reveal credential features

## **Related issues**

Fixes: NA

## **Manual testing steps**

**Test mode**

1. Build the app in dev or test mode
2. Go to the Security & Privacy settings view
3. Try to reveal the SRP and/or Private Key
4. Verify that the "Hold to reveal" button is no longer present

**Prod mode**

1. Build the app in prod mode
2. Go to the Security & Privacy settings view
3. Try to reveal the SRP and/or Private Key
4. Verify that the "Hold to reveal" button is present

### **After**

| Dev/Test Mode | Prod mode |
|---------------|-----------|
|  |            |

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by cortisiko — DISMISSED** [review]
Thanks, Guto! The approach of using a flag to disable the animation for end-to-end testing makes sense to me, as Detox has issues interacting with animated buttons.

File: app/components/Views/RevealPrivateCredential/RevealPrivateCredential.tsx [hunk 1]
```diff
@@ -458,18 +462,28 @@ const RevealPrivateCredential = ({
...
+              style={styles.revealButton}
+            />
+          ) : (
+            <ButtonReveal
+              label={strings('reveal_credential.hold_to_reveal_credential', {
+                credentialName: isPrivateKeyReveal
+                  ? strings('reveal_credential.private_key_text')
+                  : strings('reveal_credential.srp_abbreviation_text'),
+              })}
+              onLongPress={() => revealCredential(privCredentialName)}
```
Comment (owencraston) [inline app/components/Views/RevealPrivateCredential/RevealPrivateCredential.tsx:484]: Nit: these callbacks could be memoized with `useCallback`

**Review by owencraston — DISMISSED** [review]
Ran through a local test and the behaviour was what I expected. Code looks good to me!

File: app/components/Views/RevealPrivateCredential/RevealPrivateCredential.tsx (hunk 1 above)
Comment (gantunesr) [reply app/components/Views/RevealPrivateCredential/RevealPrivateCredential.tsx]: Done in [7aed81f](https://github.com/MetaMask/metamask-mobile/pull/10608/commits/7aed81f1136b7423acadc7146005b13413e69a58)

File: app/components/Views/RevealPrivateCredential/index.test.tsx [hunk 2]
```diff
@@ -23,21 +25,95 @@ const initialState = {
...
+            credentialName: SRP_CREDENTIAL,
+          },
+        }}
+        navigation={null}
+        cancel={() => null}
+        credentialName={SRP_CREDENTIAL}
+      />,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
```
Comment (montelaidev) [inline app/components/Views/RevealPrivateCredential/index.test.tsx:72]: I think we can create additional tests where `route.params.credentialName` or `credentialName` is undefined to ensure that we are still able to get to the reveal page.

Comment (gantunesr) [reply app/components/Views/RevealPrivateCredential/index.test.tsx]: Done in [e03f434](https://github.com/MetaMask/metamask-mobile/pull/10608/commits/e03f4348403777f31a30b14c971c568870deb0ca)

**Review by owencraston — APPROVED** [review]

**Review by tommasini — APPROVED** [review]

**Pull request merged**